### PR TITLE
Pair plot fixes and kde extra options. Fixes #602

### LINKED ISSUE #602: Change pair_plot behavior for MCMC methods that do not produce divergences
**Describe the bug**
When using metropolis hasting sampler in pymc3, then converting to inference data, then using `az.pair_plot(divergences=True)` raises an index error

**To Reproduce**
```
import matplotlib.pyplot as plt
import arviz as az
import pymc3 as pm

x = [1,2]
y = [1,2]
with pm.Model() as model_g:
    α = pm.Normal('α', mu=0, sd=10)
    β = pm.Normal('β', mu=0, sd=1)
    ϵ = pm.HalfCauchy('ϵ', 5)

    μ = pm.Deterministic('μ', α + β * x)
    y_pred = pm.Normal('y_pred', mu=μ, sd=ϵ, observed=y)

# NUTS sampler works when plotting divergences
"""
with model_g:
    trace_nuts_non_centered = pm.sample()

nuts_non_centered_dataset = az.from_pymc3(trace=trace_nuts_non_centered)
az.plot_pair(nuts_non_centered_dataset, var_names=['α', 'β', 'ϵ'], divergences=True)

plt.show()
"""

# Metropolis Hastings raises indexerror
with model_g:
    step = pm.Metropolis()
    trace_mh_non_centered = pm.sample(step=step)

az.plot_pair(trace_mh_non_centered, var_names=['α', 'β', 'ϵ'], divergences=True)
plt.show()
```

**Expected behavior**
~~Divergences are plotted as expected~~
Show a warning and continue plotting

**Additional context**
Versions of `arviz` and other libraries used, operating system used, and anything else that may be useful.

Arviz Version  '0.3.3'
PyMC3 Version : '3.6'

### PR DESCRIPTION
**Pull request by OriolAbril**
It should fix the issue with missing divergence data (I choose to use a `warnings.warn` instead of logging following #620) and added kwarg arguments for contourf and pcolormesh in `plot_kde`.

### CODE REVIEW

**Comment by canyon289** [comment]
Can tests be added for these?

File: arviz/plots/pairplot.py [hunk 1]
```diff
@@ -141,11 +143,20 @@ def plot_pair(
...
+        if hasattr(data, "sample_stats") and hasattr(data.sample_stats, "diverging"):
+            divergent_data = convert_to_dataset(data, group="sample_stats")
+            _, diverging_mask = xarray_to_ndarray(
+                divergent_data, var_names=("diverging",), combined=True
+            )
+            diverging_mask = np.squeeze(diverging_mask)
+        else:
+            divergences = False
+            warnings.warn(
+                "diverging field not found in sample stats. "
```
Comment (canyon289) [inline arviz/plots/pairplot.py]: Great message

File: arviz/tests/test_plots.py [hunk 2]
```diff
@@ -387,6 +401,18 @@ def test_plot_pair_bad(models, model_fit):
         plot_pair(obj, var_names=["mu"])
 
 
+@pytest.mark.parametrize("has_stats", [True, False])
+def test_plot_pair_divergences_warning(has_stats):
+    data = load_arviz_data("centered_eight")
+    if has_stats:
+        data.sample_stats = data.sample_stats.rename({"diverging": "diverging_missing"})
```
Comment (canyon289) [inline arviz/tests/test_plots.py]: So if `has_stats` is true that means we set `diverging` to `diverging_mission`?

Should this be the opposite case semantically?

Comment (OriolAbril) [reply arviz/tests/test_plots.py]: The warning can be due to 2 reasons, either the InferenceData object has no `sample_stats` group, or the `sample_stats` dataset has no `diverging` variable. In the test, `has_stats=True` indicates the second case, whereas `has_stats=False` is the first case, the input only has a posterior group. 
Maybe I should modify the message of the warning as it is basically thought for the second case.

File: arviz/plots/pairplot.py (hunk 1 above)
Comment (aloctavodia) [reply arviz/plots/pairplot.py]: I like the message, but maybe someone may think that the diverging field _should_ be there. And most likely what is happening is that the sampling method does not provide that information.

File: arviz/tests/test_plots.py [hunk 3]
```diff
@@ -272,15 +272,29 @@ def test_plot_joint_bad(models, model_fit):
     [
         {"plot_kwargs": {"linestyle": "-"}},
         {"contour": True, "fill_last": False},
+        {
+            "contour": True,
+            "contourf_kwargs": {"cmap": "plasma"},
```
Comment (aloctavodia) [inline arviz/tests/test_plots.py]: Why `plasma` and not `viridis`, that is the matplotlib default?

Comment (OriolAbril) [reply arviz/tests/test_plots.py]: To test different values, no particular reason.

Comment (aloctavodia) [reply arviz/tests/test_plots.py]: I have the impression more people like plasma than viridis...

File: arviz/plots/pairplot.py (hunk 1 above)
Comment (OriolAbril) [reply arviz/plots/pairplot.py]: What about ```"Divergences data not found, plotting without divergences. Make sure the sample method provides divergences data and that it is present in the `diverging` field of `sample_stats` or set divergences=False"```

**Comment by canyon289** [comment]
This LGTM. Will leave for a day for others to review

**Pull request merged**